Bug: Test case test_bgp_update_timer_session_down and test_bgp_update_timer_single_route failed due to #22924

### Is it platform specific

generic

### Importance or Severity

Critical

### Description of the bug

Invalid command: clear ipv6 bgp fc02:1000::2 soft out
tests.common.errors.RunAnsibleModuleFail: run module shell failed, Ansible Results =>
failed = True
changed = True
rc = 1
cmd = vtysh  -c 'clear ipv6 bgp fc02:1000::2 soft out'
start = 2026-06-09 03:31:12.801055
end = 2026-06-09 03:31:12.907155
delta = 0:00:00.106100
msg = non-zero return code
invocation = {'module_args': {'_raw_params': "vtysh  -c 'clear ipv6 bgp fc02:1000::2 soft out'", '_uses_shell': True, 'expand_argument_vars': True, 'stdin_add_newline': True, 'strip_empty_ends': True, 'argv': None, 'chdir': None, 'executable': None, 'creates': None, 'removes': None, 'stdin': None}}
_ansible_no_log = False
stdout =
% Unknown command: clear ipv6 bgp fc02:1000::2 soft outstderr =

The invalid command introduced by #22924 

### Steps to Reproduce

Run the test cases

### Actual Behavior and Expected Behavior

**Actual Behavior:**
tests.common.errors.RunAnsibleModuleFail: run module shell failed, Ansible Results =>
failed = True
changed = True
rc = 1
cmd = vtysh  -c 'clear ipv6 bgp fc02:1000::2 soft out'
start = 2026-06-09 03:31:12.801055
end = 2026-06-09 03:31:12.907155
delta = 0:00:00.106100
msg = non-zero return code
invocation = {'module_args': {'_raw_params': "vtysh  -c 'clear ipv6 bgp fc02:1000::2 soft out'", '_uses_shell': True, 'expand_argument_vars': True, 'stdin_add_newline': True, 'strip_empty_ends': True, 'argv': None, 'chdir': None, 'executable': None, 'creates': None, 'removes': None, 'stdin': None}}
_ansible_no_log = False
stdout =
% Unknown command: clear ipv6 bgp fc02:1000::2 soft outstderr =

**Expected Behavior**: The test case should pass.

### Relevant log output

```shell
tests.common.errors.RunAnsibleModuleFail: run module shell failed, Ansible Results =>
failed = True
changed = True
rc = 1
cmd = vtysh  -c 'clear ipv6 bgp fc02:1000::2 soft out'
start = 2026-06-09 03:31:12.801055
end = 2026-06-09 03:31:12.907155
delta = 0:00:00.106100
msg = non-zero return code
invocation = {'module_args': {'_raw_params': "vtysh  -c 'clear ipv6 bgp fc02:1000::2 soft out'", '_uses_shell': True, 'expand_argument_vars': True, 'stdin_add_newline': True, 'strip_empty_ends': True, 'argv': None, 'chdir': None, 'executable': None, 'creates': None, 'removes': None, 'stdin': None}}
_ansible_no_log = False
stdout =
% Unknown command: clear ipv6 bgp fc02:1000::2 soft outstderr =
common_setup_teardown = ((, ), False)
constants = ._C object at 0x7fdb1d8241a0>
duthosts = []
enum_rand_one_per_hwsku_frontend_hostname = 'r-bison-16'
request = >
toggle_all_simulator_ports_to_enum_rand_one_per_hwsku_frontend_host_m = None
validate_active_active_dualtor_setup = None
tbinfo = {'comment': 'Mellanox MTR testbed', 'conf-name': 'r-bison-16-t0-isolated-v6-d32u32s2', 'duts': ['r-bison-16'], 'duts_map': {'r-bison-16': 0}, ...}

    def test_bgp_update_timer_single_route(
        common_setup_teardown,
        constants,
        duthosts,
        enum_rand_one_per_hwsku_frontend_hostname,
        request,
        toggle_all_simulator_ports_to_enum_rand_one_per_hwsku_frontend_host_m,  # noqa:F811
        validate_active_active_dualtor_setup,  # noqa:F811
        tbinfo
    ):
        duthost = duthosts[enum_rand_one_per_hwsku_frontend_hostname]
        is_v6_topo = is_ipv6_only_topology(tbinfo)
    
        (n0, n1), _ = common_setup_teardown
        try:
            n0.start_session()
            n1.start_session()
            bgp_neighbor_cmd_tmpl = (
                "show bgp ipv6 neighbors {} received-routes"
                if is_v6_topo
                else "show ip bgp neighbors {} received-routes"
            )
    
            # ensure new sessions are ready
            if not wait_until(
                WAIT_TIMEOUT,
                5,
                20,
                lambda: is_neighbor_sessions_established(duthost, (n0, n1)),
            ):
                pytest.fail("Could not establish bgp sessions")
    
            # Restrict outbound routes to ExaBGP neighbors to only allow test
            # prefixes, preventing the DUT from flooding them with the full
            # routing table which causes session flapping (issue #22391)
>           _apply_outbound_route_filter(
                duthost, n0.peer_asn, [n0.ip, n1.ip], is_v6_topo, n0.namespace
            )

_          = False
bgp_neighbor_cmd_tmpl = 'show bgp ipv6 neighbors {} received-routes'
common_setup_teardown = ((, ), False)
constants  = ._C object at 0x7fdb1d8241a0>
duthost    = 
duthosts   = []
enum_rand_one_per_hwsku_frontend_hostname = 'r-bison-16'
is_v6_topo = True
n0         = 
n1         = 
request    = >
route      = {'nexthop': 'fc02:1000::2', 'prefix': 'fc00:14::/64'}
tbinfo     = {'comment': 'Mellanox MTR testbed', 'conf-name': 'r-bison-16-t0-isolated-v6-d32u32s2', 'duts': ['r-bison-16'], 'duts_map': {'r-bison-16': 0}, ...}
toggle_all_simulator_ports_to_enum_rand_one_per_hwsku_frontend_host_m = None
validate_active_active_dualtor_setup = None

bgp/test_bgp_update_timer.py:431: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
bgp/test_bgp_update_timer.py:104: in _apply_outbound_route_filter
    duthost.shell("vtysh {} -c 'clear {} bgp {} soft out'".format(
        cmd        = "vtysh  -c 'configure terminal' -c 'ipv6 prefix-list PL_BGP_UPDATE_TEST seq 10 permit fc00:10::/44 le 64' -c 'route-ma...0::2 route-map RM_BGP_UPDATE_TEST out' -c 'neighbor fc02:1000::3 route-map RM_BGP_UPDATE_TEST out' -c 'exit' -c 'exit'"
        dut_asn    = 4200000000
        duthost    = 
        ip         = 'fc02:1000::2'
        is_v6      = True
        namespace  = None
        neighbor_ips = ['fc02:1000::2', 'fc02:1000::3']
        ns_option  = ''
        pfx_kw     = 'ipv6'
        prefix_match = 'fc00:10::/44 le 64'
        vtysh_cmds = ['configure terminal', 'ipv6 prefix-list PL_BGP_UPDATE_TEST seq 10 permit fc00:10::/44 le 64', 'route-map RM_BGP_UPDAT...T permit 10', 'match ipv6 address prefix-list PL_BGP_UPDATE_TEST', 'exit', 'route-map RM_BGP_UPDATE_TEST deny 20', ...]
common/devices/multi_asic.py:403: in _run_on_asics_wrapper
    return self._run_on_asics(attr, *module_args, **complex_args)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        attr       = 'shell'
        complex_args = {}
        module_args = ("vtysh  -c 'clear ipv6 bgp fc02:1000::2 soft out'",)
        self       = 
common/devices/multi_asic.py:160: in _run_on_asics
    return getattr(self.sonichost, multi_asic_attr)(*module_args, **complex_args)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        complex_args = {}
        module_args = ("vtysh  -c 'clear ipv6 bgp fc02:1000::2 soft out'",)
        multi_asic_attr = 'shell'
        self       = 
common/devices/base.py:163: in _run_wrapper
    return self._run(module_name, *module_args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        kwargs     = {}
        module_args = ("vtysh  -c 'clear ipv6 bgp fc02:1000::2 soft out'",)
        module_name = 'shell'
        self       = 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = , module_name = 'shell'
module_args = ["vtysh  -c 'clear ipv6 bgp fc02:1000::2 soft out'"]
complex_args = {}
previous_frame = 
filename = '/root/mars/workspace/sonic-mgmt/tests/common/devices/base.py'
line_number = 163, function_name = '_run_wrapper'
lines = ['                return self._run(module_name, *module_args, **kwargs)\n']
index = 0, verbose = True, module_ignore_errors = False, module_async = False

    def _run(self, module_name, *module_args, **complex_args):
    
        previous_frame = inspect.currentframe().f_back
        filename, line_number, function_name, lines, index = inspect.getframeinfo(previous_frame)
    
        verbose = complex_args.pop('verbose', True)
        module = getattr(self.host, module_name)
        if verbose:
            logger.debug(
                "{}::{}#{}: [{}] AnsibleModule::{}, args={}, kwargs={}".format(
                    filename,
                    function_name,
                    line_number,
                    self.hostname,
                    module_name,
                    json.dumps(module_args, cls=AnsibleHostBase.CustomEncoder),
                    json.dumps(complex_args, cls=AnsibleHostBase.CustomEncoder)
                )
            )
        else:
            logger.debug(
                "{}::{}#{}: [{}] AnsibleModule::{} executing...".format(
                    filename,
                    function_name,
                    line_number,
                    self.hostname,
                    module_name
                )
            )
    
        module_ignore_errors = complex_args.pop('module_ignore_errors', False)
        module_async = complex_args.pop('module_async', False)
    
        if module_async:
            def run_module(module_args, complex_args):
                with suppress_signal_registration_for_non_main_thread():
                    return module(*module_args, **complex_args)[self.hostname]
            pool = ThreadPool()
            result = pool.apply_async(run_module, (module_args, complex_args))
            return pool, result
    
        module_args = json.loads(json.dumps(module_args, cls=AnsibleHostBase.CustomEncoder))
        complex_args = json.loads(json.dumps(complex_args, cls=AnsibleHostBase.CustomEncoder))
    
        with suppress_signal_registration_for_non_main_thread():
            adhoc_res: AdHocResult = module(*module_args, **complex_args)
    
        if module_name == "meta":
            # The meta module is special in Ansible - it doesn't execute on remote hosts, it controls Ansible's behavior
            # There are no per-host ModuleResults contained within it
            return
    
        hostname_res: ModuleResult = adhoc_res[self.hostname]
        hostname_res.encoder = AnsibleHostBase.CustomEncoder
    
        if verbose:
            logger.debug(
                "{}::{}#{}: [{}] AnsibleModule::{} Result => {}".format(
                    filename,
                    function_name,
                    line_number,
                    self.hostname,
                    module_name, json.dumps(hostname_res, cls=AnsibleHostBase.CustomEncoder)
                )
            )
        else:
            logger.debug(
                "{}::{}#{}: [{}] AnsibleModule::{} done, is_failed={}, rc={}".format(
                    filename,
                    function_name,
                    line_number,
                    self.hostname,
                    module_name,
                    hostname_res.is_failed,
                    hostname_res.get('rc', None)
                )
            )
    
        if (hostname_res.is_failed or 'exception' in hostname_res) and not module_ignore_errors:
>           raise RunAnsibleModuleFail("run module {} failed".format(module_name), hostname_res)
E           tests.common.errors.RunAnsibleModuleFail: run module shell failed, Ansible Results =>
E           failed = True
E           changed = True
E           rc = 1
E           cmd = vtysh  -c 'clear ipv6 bgp fc02:1000::2 soft out'
E           start = 2026-06-09 03:31:12.801055
E           end = 2026-06-09 03:31:12.907155
E           delta = 0:00:00.106100
E           msg = non-zero return code
E           invocation = {'module_args': {'_raw_params': "vtysh  -c 'clear ipv6 bgp fc02:1000::2 soft out'", '_uses_shell': True, 'expand_argument_vars': True, 'stdin_add_newline': True, 'strip_empty_ends': True, 'argv': None, 'chdir': None, 'executable': None, 'creates': None, 'removes': None, 'stdin': None}}
E           _ansible_no_log = False
E           stdout =
E           % Unknown command: clear ipv6 bgp fc02:1000::2 soft outstderr =

adhoc_res  = 
complex_args = {}
filename   = '/root/mars/workspace/sonic-mgmt/tests/common/devices/base.py'
function_name = '_run_wrapper'
hostname_res = {'failed': True, 'changed': True, 'stdout': '% Unknown command: clear ipv6 bgp fc02:1000::2 soft out', 'stderr': '', '...dout_lines': ['% Unknown command: clear ipv6 bgp fc02:1000::2 soft out'], 'stderr_lines': [], '_ansible_no_log': False}
index      = 0
line_number = 163
lines      = ['                return self._run(module_name, *module_args, **kwargs)\n']
module     = >
module_args = ["vtysh  -c 'clear ipv6 bgp fc02:1000::2 soft out'"]
module_async = False
module_ignore_errors = False
module_name = 'shell'
previous_frame = 
self       = 
verbose    = True

common/devices/base.py:248: RunAnsibleModuleFail

```

### Output of `show version`

```shell
SONiC Software Version: SONiC.master_RC_integration.21-8c949ab73_Internal
SONiC OS Version: 13
Distribution: Debian 13.3
Kernel: 6.12.41+deb13-sonic-amd64
Build commit: 19de3f150
Build date: Mon Jun  8 05:27:39 UTC 2026
Built by: sw-r2d2-bot@r-build-sonic-ci03-244
Platform: x86_64-nvidia_sn5640-r0
HwSKU: Mellanox-SN5640-C512S2
ASIC: mellanox
ASIC Count: 1
Serial Number: MT2514601H7E
Model Number: MSN5640
Hardware Revision: A5
Uptime: 03:47:22 up  3:05,  4 users,  load average: 2.30, 2.31, 2.20
Date: Tue 09 Jun 2026 03:47:22
```

### Attach files (if any)

_No response_

--- a/tests/bgp/test_bgp_update_timer.py
+++ b/tests/bgp/test_bgp_update_timer.py
@@ -99,10 +99,14 @@ def _apply_outbound_route_filter(duthost, dut_asn, neighbor_ips, is_v6, namespac
     cmd = "vtysh {} {}".format(ns_option, " ".join("-c '{}'".format(c) for c in vtysh_cmds))
     duthost.shell(cmd)
 
-    # Soft-reset outbound so the filter takes effect immediately
+    # Soft-reset outbound so the filter takes effect immediately.
+    # Note: FRR vtysh syntax differs between v4 and v6:
+    #   v4: clear ip bgp <neighbor> soft out
+    #   v6: clear bgp ipv6 <neighbor> soft out  (word order is 'bgp ipv6', not 'ipv6 bgp')
+    clear_af = "bgp ipv6" if is_v6 else "ip bgp"
     for ip in neighbor_ips:
-        duthost.shell("vtysh {} -c 'clear {} bgp {} soft out'".format(
-            ns_option, "ipv6" if is_v6 else "ip", ip
+        duthost.shell("vtysh {} -c 'clear {} {} soft out'".format(
+            ns_option, clear_af, ip
         ))
 
 

